sycl/test/check_device_code/device_has.cpp

```diff
@@ -45,29 +44,19 @@ constexpr sycl::aspect getAspect() { return sycl::aspect::cpu; }
 // CHECK-SRCLOC: define {{.*}}spir_func void @{{.*}}func6{{.*}} !srcloc ![[SRCLOC7:[0-9]+]]
 [[sycl::device_has(getAspect())]] void func6() {}
 
-class KernelFunctor {
-public:
-  [[sycl::device_has(sycl::aspect::cpu)]] void operator()() const {
-    func1();
-    func2();
-    func3();
-    func4<sycl::aspect::host>();
-    func5();
-    func6();
-  }
-};
-
-void foo() {
-  q.submit([&](handler &h) {
-    KernelFunctor f1;
-    h.single_task<class kernel_name_1>(f1);
-    // CHECK-ASPECTS: define weak_odr dso_local spir_kernel void @{{.*}}kernel_name_2{{.*}} !sycl_declared_aspects ![[ASPECTS4:[0-9]+]]
-    // CHECK-SRCLOC: define weak_odr dso_local spir_kernel void @{{.*}}kernel_name_2{{.*}} !srcloc ![[SRCLOC8:[0-9]+]] {{.*}}
-    h.single_task<class kernel_name_2>(
-        []() [[sycl::device_has(sycl::aspect::gpu)]] {});
-  });
+SYCL_EXTERNAL [[sycl::device_has(sycl::aspect::cpu)]] void kernel_name_1() {
+  func1();
+  func2();
+  func3();
+  func4<sycl::aspect::host>();
+  func5();
+  func6();
 }
 
+// CHECK-ASPECTS: define dso_local spir_func void @{{.*}}kernel_name_2{{.*}} !sycl_declared_aspects ![[ASPECTS4:[0-9]+]]
+// CHECK-SRCLOC: define dso_local spir_func void @{{.*}}kernel_name_2{{.*}} !srcloc ![[SRCLOC8:[0-9]+]] {{.*}}
+SYCL_EXTERNAL [[sycl::device_has(sycl::aspect::gpu)]] void kernel_name_2() {}
```

Comment:
I think we might need *both* new and old code, actually. Someone from @intel/dpcpp-tools-reviewers should review changes in this file.

Reply:
I split the test cases into 2 files, one for kernels and one for functions. Am I good to proceed with this?